New anointing/dev

lolade-dev opened the pull request.

New-Anointing committed:
changed http responses and added get all users and get all users by id

New-Anointing committed:
pushed subjects actions

New-Anointing committed:
TimeTablefunctions

lolade-dev commented on SchoolManagementSystemApi/Controllers/RegistrationController.cs:
```diff
@@ -44,5 +44,25 @@ public async Task<ActionResult> UserRegistration(UserDTO request)
+        [Authorize(Roles = SD.Admin +","+ SD.SuperAdmin)]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(GenericResponse<ApplicationUser>))]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError, Type=typeof(GenericResponse<ApplicationUser>))]
+        [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(GenericResponse<Exception>))]   
```
@New-Anointing Is there a reason why "Exception" is used here?

New-Anointing replied:
that was a mistake will correct that now

lolade-dev commented on SchoolManagementSystemApi/Model/Teachers.cs:
```diff
@@ -1,17 +1,17 @@
 {
     public class Teachers
     {
-        public Guid Id { get; set; }
```
@New-Anointing Why was this deleted?

lolade-dev commented on SchoolManagementSystemApi/Model/Teachers.cs:
```diff
@@ -1,17 +1,17 @@
-        public string salary { get; set; }
+        public Guid UserId { get; set; }
+        [ForeignKey("UserId")]
+        public virtual ApplicationUser ApplicationUser { get; set; }
```
@New-Anointing I thought we came to a conclusion not to use this. Even if we are, why are there two instances of UserId? Same with classroomid

New-Anointing replied:
i started this before so i decided to remove it

lolade-dev commented on SchoolManagementSystemApi/Model/Subjects.cs:
```diff
@@ -0,0 +1,12 @@
+﻿using System.ComponentModel.DataAnnotations;
```
@New-Anointing Is there a reason why you removed the shortcode, your model is looking scanty. The property could have been called course code or something

New-Anointing replied:
didn't add one before i will add that now

lolade-dev commented on SchoolManagementSystemApi/Model/TimeTable.cs:
```diff
@@ -0,0 +1,21 @@
+﻿using System.ComponentModel.DataAnnotations;
```
@New-Anointing As mentioned before, you don't need both a virtual property and a new property repeated

New-Anointing committed:
resolved issues

New-Anointing committed:
fixed timetable foreign key refrence

New-Anointing committed:
added teachers models

Commit:
Some corrections

New-Anointing committed:
commit for events functions

New-Anointing committed:
complrtion for events

New-Anointing committed:
Changes To the Time table controller and Events controller and models
created students and teachers model

New-Anointing committed:
commit for teachers services

New-Anointing committed:
commit for students services
-GetAllStudents
-RegisterStudents
-GetAllRegisteredStudents

New-Anointing committed:
Commit For
-Edit endpoint for Event Services
-creation of class teacher endpoint

New-Anointing committed:
commit for
-Adding list of subjects to teachers
-incude students query for students

New-Anointing committed:
Commit For
-get registered students by id
-assign student class
-assign students subjects
-roles database seed

New-Anointing committed:
commit for
-initilization of parents endpoints

New-Anointing committed:
Merge pull request #8 from New-Anointing/master

Merge pull request #6 from New-Anointing/New-Anointing/dev

New-Anointing committed:
commit for
-parent services

New-Anointing committed:
commit for
-all parent api end points
